elements/bulbs-pinned-element/bulbs-pinned-element.js

```diff
@@ -1,128 +1,133 @@
-import { registerElement, BulbsHTMLElement } from 'bulbs-elements/register';
 import invariant from 'invariant';
-import { getScrollOffset, InViewMonitor } from 'bulbs-elements/util';
+
+import {
+  registerElement,
+  BulbsHTMLElement
+} from 'bulbs-elements/register';
+import {
+  moveChildren,
+  getScrollOffset,
+  InViewMonitor
+} from 'bulbs-elements/util';
 import './bulbs-pinned-element.scss';
 
 export default class BulbsPinnedElement extends BulbsHTMLElement {
+
   attachedCallback () {
-    invariant(this.hasAttribute('pinned-to'), '<bulbs-pinned-element pinned-to=".selector">: a pinned-to selector is required');
 
-    const selector = this.getAttribute('pinned-to');
-    const element = document.querySelector(selector);
+    this.topOffsetAdjustment = parseInt(this.getAttribute('offset-top-px') || 0, 10);
     this.lastPosition = 0;
     this.animationRequest = null;
 
-    invariant(element, `<bulbs-pinned-element pinned-to=".selector">: no element with the selector "${selector}" is in the DOM`);
+    let car = this.querySelector('bulbs-pinned-element-car');
```

Comment:
This child element is provided by the `bulbs-pinned-element` itself in the following lines. Here I'm mirroring the defensive logic contained in some other components to keep a reference to existing components if the element is unmounted and remounted by external logic.

I will update the readme for this component before review and release.